middleware/src/middleware.go

```diff
@@ -597,3 +561,89 @@ func (middleware *Middleware) SetRootPassword(args rpcmessages.SetRootPasswordAr
 		Code:    rpcmessages.ErrorSetRootPasswordTooShort,
 	}
 }
+
+// GetBaseInfo returns information about the Base in a GetBaseInfoResponse
+func (middleware *Middleware) GetBaseInfo() rpcmessages.GetBaseInfoResponse {
+	const (
+		// Prometheus queries
+		freeDiskspaceQuery  string = "node_filesystem_free_bytes{fstype=\"ext4\", mountpoint=\"/mnt/ssd\"}"
+		totalDiskspaceQuery string = "node_filesystem_size_bytes{fstype=\"ext4\", mountpoint=\"/mnt/ssd\"}"
+	)
+
+	hostname, errResp := middleware.getStringFromRedis(redis.BaseHostname)
+	if !errResp.Success {
+		return rpcmessages.GetBaseInfoResponse{ErrorResponse: &errResp}
+	}
+
+	middlewareIP, errResp := middleware.getBaseIPFromPrometheus()
+	if !errResp.Success {
+		return rpcmessages.GetBaseInfoResponse{ErrorResponse: &errResp}
+	}
+
+	middlewarePort := middleware.environment.GetMiddlewarePort()
+
+	middlewareTorOnion, errResp := middleware.getStringFromRedis(redis.MiddlewareOnion)
+	if !errResp.Success {
+		return rpcmessages.GetBaseInfoResponse{ErrorResponse: &errResp}
+	}
+
+	isTorEnabled, errResp := middleware.getBooleanFromRedis(redis.TorEnabled)
+	if !errResp.Success {
+		return rpcmessages.GetBaseInfoResponse{ErrorResponse: &errResp}
+	}
+
+	isBitcoindListening, errResp := middleware.getBooleanFromRedis(redis.BitcoindListen)
+	if !errResp.Success {
+		return rpcmessages.GetBaseInfoResponse{ErrorResponse: &errResp}
+	}
+
+	freeDiskspace, errResp := middleware.getIntFromPrometheus(freeDiskspaceQuery)
+	if !errResp.Success {
+		return rpcmessages.GetBaseInfoResponse{ErrorResponse: &errResp}
+	}
+
+	totalDiskspace, errResp := middleware.getIntFromPrometheus(totalDiskspaceQuery)
+	if !errResp.Success {
+		return rpcmessages.GetBaseInfoResponse{ErrorResponse: &errResp}
+	}
+
+	baseVersion, errResp := middleware.getStringFromRedis(redis.BaseVersion)
+	if !errResp.Success {
+		return rpcmessages.GetBaseInfoResponse{ErrorResponse: &errResp}
+	}
+
+	bitcoindVersion, errResp := middleware.getStringFromRedis(redis.BitcoindVersion)
+	if !errResp.Success {
+		return rpcmessages.GetBaseInfoResponse{ErrorResponse: &errResp}
+	}
+
+	lightningdVersion, errResp := middleware.getStringFromRedis(redis.LightningdVersion)
+	if !errResp.Success {
+		return rpcmessages.GetBaseInfoResponse{ErrorResponse: &errResp}
+	}
+
+	electrsVersion, errResp := middleware.getStringFromRedis(redis.ElectrsVersion)
+	if !errResp.Success {
+		return rpcmessages.GetBaseInfoResponse{ErrorResponse: &errResp}
+	}
+
+	return rpcmessages.GetBaseInfoResponse{
+		ErrorResponse: &rpcmessages.ErrorResponse{
+			Success: true,
```

Comment:
unrelated, but: Success flag looks redundant. ErrorResponse could be `nil` when no error, and then it looks a bit more like Go `if response.Error  != nil { return response.Error } `)

Reply:
Not addressing this, but I'll keep it in mind.